app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+
+        // TODO Is this the way that we should be specifying race? It's required in the API, but not in the app
+        val race = (additionalInfoData.race ?: Race.DECLINE).toString()
+
+        val party = when (additionalInfoData.party) {
+            Party.OTHER_PARTY -> additionalInfoData.otherPoliticalParty
+                ?: throw InvalidRegistrationException(
+                    "otherPoliticalParty is null",
+                    R.string.must_write_in_party
+                )
+            else -> additionalInfoData.party.toString()
+        }.toLowerCase() // TODO i
+
+        // TODO NO_WRAP was specified in ApiSignature, but probably should be in the API docs as well
+        val signature = Base64.encodeToString(reviewData.signature, Base64.NO_WRAP)
+
+        return VoterRegistration(
+            registrationHelper = buildRegistrationHelper(),
+            dateOfBirth = Dates.formatAsISO8601_ShortDate(newRegistrationData.birthday), //TODO DateAdapter used short date, is this correct?
+            mailingAddress = mailingAddress?.toApiAddressData(),
+            previousRegistrationAddress = addressData.previousAddress?.toApiAddressData(),
+            registrationAddress = addressData.homeAddress.toApiAddressData(),
+            registrationAddressIsMailingAddress = !addressData.isMailingAddressDifferent,
+            name = newRegistrationData.name.toApiName(),
+            previousName = newRegistrationData.previousName?.toApiName(),
+            gender = Gender.fromPrefix(newRegistrationData.name.title).toString(),
+            race = race, // TODO race is required in the API, but not in the app?
+            party = party,
+            voterClassifications = buildVoterClassifications(),
+            signature = Signature(
+                mimeType = "image/png",
+                image = signature
+            ),
+            voterIds = buildVoterIds(),
+            contactMethods = buildContactMethods(),
+            additionalInfo = buildAdditionalInfo()
+        )
+
+    }
+
+    private fun buildRegistrationHelper(): RegistrationHelper? {
+        if (!assistanceData.hasSomeoneAssisted) return null // TODO is this correct, or should this be based on the canvasser and always non-null?
+
+        val helperPhone = assistanceData.helperPhone ?: throw InvalidRegistrationException(
+            "helperPhone is null",
+            R.string.assistants_phone_number_is_empty
+        )
+
+        return RegistrationHelper(
+            registrationHelperType = "assistant", // TODO is this a default value?
+            name = assistanceData.helperName?.toApiName(),
+            address = assistanceData.helperAddress?.toApiAddressData(),
+            contactMethods = listOf(
+                ContactMethod(
+                    type = "phone",
+                    value = helperPhone,
+                    capabilities = listOf("voice") // TODO should capabilties be more here? IIRC, this is only VOICE from [ApiContactMethod]
```

Comment:
see the previous comment on phone type for a more detailed explanation, but basically if it's a cell phone it gets "sms and voice" if it's a different phone it just gets "voice"

Reply:
I think my initial confusion was around the UX that this is derived from doesn't have a selection of phone type, it just requires a phone number alone.

Looking at this more though, it looks like this is actually a `type` `assistant_phone` in the previous implementation, which means `voice` alone would be correct.